src/Uno.UI/UI/Xaml/Controls/Slider/Slider.cs

```diff
@@ -321,6 +362,59 @@ private double GetVerticalThumbHeight()
 			return _verticalThumb.ActualHeight;
 		}
 
+		/// <summary>
+		/// Calculate the max width
+		/// </summary>
+		/// <remarks>
+		/// If the thumb is as big as the control,
+		/// the thumb should be able to move but only a little bit to mimic UWP behavior.
+		/// </remarks>
+		/// <param name="sliderWidth">The slider height. Needed to determine if the thumb should move or not.</param>
+		/// <param name="thumbWidth">The thumb height. Needed to determine if the thumb should move or not.</param>
+		private double CalculateMaxWidth(double sliderWidth, double thumbWidth)
+		{
+			var result = sliderWidth - thumbWidth;
+			var maxWidthShouldBeFixed = (result <= 0);
+			if (maxWidthShouldBeFixed)
+			{
+				result = 0.1;
+			}
+
+			return result;
+		}
+
+		/// <summary>
+		/// Calculate the controls max height
+		/// </summary>
+		/// <remarks>
+		/// If the thumb is as big as the control,
+		/// the thumb should be able to move but only a little bit to mimic UWP behavior.
+		/// </remarks>
+		/// <param name="sliderHeight">The slider height. Needed to determine if the thumb should move or not.</param>
+		/// <param name="thumbHeight">The thumb height. Needed to determine if the thumb should move or not.</param>
+		private double CalculateMaxHeight(double sliderHeight, double thumbHeight)
+		{
+			var result = sliderHeight - thumbHeight;
+			var acceptableHeightDifference = 0;
+
+			// On android and ios, when the thumb is the same size as the control or bigger
+			// it causes the control to stretch verticaly. The actual height of the thumb
+			// is not reliable and it often changes. The value of "5" for the acceptable difference
+			// was obtain from multiple test but might not work for every cases. This is obviously
+			// a work around has the issue is in either in the grid that makes up the slider or
+			// in the measure or arrange phase.
+#if __ANDROID__ || __IOS__
```

Comment:
Since you mentioned this might not work for every cases, I think it would be better to put this variable into [https://github.com/unoplatform/uno/blob/master/src/Uno.UI/FeatureConfiguration.cs], so that it can be adjusted without having to uno, if needed.